model-providers/vertex-ai-gemini/runtime/src/main/java/io/quarkiverse/langchain4j/vertexai/runtime/gemini/VertxAiGeminiRestApi.java

```diff
@@ -140,12 +146,19 @@ public void filter(ResteasyReactiveClientRequestContext context) {
                 @Override
                 public void run() {
                     try {
-                        context.getHeaders().add("Authorization", "Bearer " + authProvider.getBearerToken());
+                        context.getHeaders().add("Authorization", "Bearer " + getBearerToken());
                         context.resume();
                     } catch (Exception e) {
                         context.resume(e);
                     }
                 }
+
+                private String getBearerToken() {
+                    if (tokenCredential.isResolvable() && tokenCredential.get().getToken() != null) {
```

Comment:
@michalvavrik Hi Michal, I'm getting `RequestContext not active error` here and I totally fail to see why. I wrote a local test in Quarkus where, in Quarkus REST endpoint, I have a REST client with a token propagation filter accessing injected `TokenCredential` exactly the same way (`integration-tests/rest-client-oidec-token-propagation`).
I tried to use `VertxAiGeminiRestApi` directly in `PoemResource`, bypassing `PoemAiService` indirection completely, still the same problem.
Does it ring any bell to a CDI god like yourself ?

Reply:
The stack trace is interesting,
```
jakarta.enterprise.context.ContextNotActiveException: RequestScoped context was not active when trying to obtain a bean instance for a client proxy of PRODUCER_METHOD bean [class=io.quarkus.oidc.runtime.OidcTokenCredentialProducer, id=V5K1EO2JY9r4ydcnBmw5DLDRuaw]
	- you can activate the request context for a specific method using the @ActivateRequestContext interceptor binding
	at io.quarkus.arc.impl.ClientProxies.notActive(ClientProxies.java:70)
	at io.quarkus.arc.impl.ClientProxies.getSingleContextDelegate(ClientProxies.java:30)
	at io.quarkus.oidc.OidcTokenCredentialProducer_ProducerMethod_currentAccessToken_LqZ323vwEzgoIJymPKn-DMYcc6w_ClientProxy.arc$delegate(Unknown Source)
	at io.quarkus.oidc.OidcTokenCredentialProducer_ProducerMethod_currentAccessToken_LqZ323vwEzgoIJymPKn-DMYcc6w_ClientProxy.getToken(Unknown Source)
	at io.quarkiverse.langchain4j.vertexai.runtime.gemini.VertxAiGeminiRestApi$TokenFilter$1.getBearerToken(VertxAiGeminiRestApi.java:157)
	at io.quarkiverse.langchain4j.vertexai.runtime.gemini.VertxAiGeminiRestApi$TokenFilter$1.run(VertxAiGeminiRestApi.java:149)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:599)
```

Executor is used to run the filter, I wonder if the request context is not bound to it

Reply:
Let me try to move the injected credential check outside of Runnable is it is just a var check

Reply:
That seemed to fix it. I should've looked more carefully at the trace